internal/pubsub/pubsub.go

```diff
@@ -193,23 +188,28 @@ func (s *Subscription) Close() error {
 	return s.driver.Close()
 }
 
+// ackIDBox makes it possible to use a driver.AckID with bundler.
+type ackIDBox struct {
+	ackID driver.AckID
+}
+
 // NewSubscription creates a Subscription from a driver.Subscription and opts to
 // tune sending and receiving of acks and messages. Behind the scenes,
 // NewSubscription spins up a goroutine to gather acks into batches and
 // periodically send them to the server.
 // It is for use by provider implementations.
 func NewSubscription(d driver.Subscription) *Subscription {
 	handler := func(item interface{}) {
-		ms := item.([]*Message)
+		boxes := item.([]ackIDBox)
 		var ids []driver.AckID
-		for _, m := range ms {
-			id := m.ackID
+		for _, box := range boxes {
+			id := box.ackID
```

Comment:
Just inline `box.ackID` below.

Reply:
Wish I could.

Reply:
I'm sorry. You're saying that what you have compiles, but
```
ids = append(ids, box.ackID)
```
does not? Can you tell me what error you get from that line?

Reply:
There's a closure that refers to id. If it refers to the loop variable then it will get the wrong value sometimes. It will still compile.

Reply:
Ah, right, thanks.